calico_kubernetes/calico_kubernetes.py

```diff
@@ -726,15 +728,55 @@ def run_protected():
     """
     Runs the plugin, intercepting all exceptions.
     """
-    # Configure logging
-    configure_logger(logger, LOG_LEVEL, True)
-    configure_logger(pycalico_logger, LOG_LEVEL, False)
+    # Parse arguments and configure logging
+    global logger
+    mode = sys.argv[1]
+
+    # Append a stdout logging handler to log to the Kubelet.
+    # We cannot do this in the status hook because the Kubelet looks to
+    # stdout for status results.
+    if mode != 'status':
+        configure_stdout_logger(logger=logger,
+                                log_level=logging.INFO,
+                                root_logger=True)
+        configure_stdout_logger(logger=pycalico_logger,
+                                log_level=logging.INFO,
+                                root_logger=False)
+
+    # Adapt the logger to append the Docker ID to logs.
+    # Docker ID is not supplied in `init`, so do not include the it in logs.
+    if mode != 'init':
+        namespace = sys.argv[2].replace('/', '_')
```

Comment:
I'd rather you did all the arg parsing in one place, at the top, and then did `if docker_id:` here; keeps the related function closer together, and makes the intent of this if/else branch cleaner.